boards/esp32*: add TTY board filter

maribu opened the pull request:
### Contribution description

This adds a TTY board filter to only select TTYs that match the attributes of the correct TTY interface. When using `make flash` or `make term` with `MOST_RECENT_PORT=1`, RIOT should now select relatively robustly the most recently connected ESP32 Ethernet Kit board, even with other TTYs present.

### Testing procedure

Connect a bunch of boards including at least one Ethernet Kit board and run

```sh
make MOST_RECENT_PORT=1 BOARD=esp32-ethernet-kit-v1_2 BUILD_IN_DOCKER=1 -j flash -C examples/default
```

This should now pick the most recently connected Ethernet Kit board. It may pick a different board when another Dual RS232-HS USB<-->UART adapter is connected, though. But those are relatively rare as integrated UART<-->USB converters, as they are (unless they are used to bit-bang JTAG) an overkill for a simple USB2UART bridge.

### Issues/PRs references

None

gschorcht commented:
:thinking: I tried it with a `esp32-wrover-kit` board as second ESP32 device which also uses the FTDI2232
```
path         | driver   | vendor       | model                                 | model_db                      | serial                           | ctime    | iface_num
-------------|----------|--------------|---------------------------------------|-------------------------------|----------------------------------|----------|----------
/dev/ttyUSB0 | cp210x   | Silicon Labs | CP2102N USB to UART Bridge Controller | CP210x UART Bridge            | e21fbb0be0faeb11bef838a4c6d924ec | 08:09:30 | 0        
/dev/ttyUSB1 | ftdi_sio | FTDI         | Dual RS232-HS                         | FT2232C/D/H Dual UART/FIFO IC | None                             | 08:12:17 | 0        
/dev/ttyUSB2 | ftdi_sio | FTDI         | Dual RS232-HS                         | FT2232C/D/H Dual UART/FIFO IC | None                             | 08:12:17 | 1        
```
using make command:
```
BOARD=esp32-ethernet-kit-v1_0 make -j8 -C examples/default/ flash
```
Although `/dev/ttyUSB2` should be used, it tries to use `/dev/ttyUSB1` to flash.
```
esptool_v3.2.py --chip esp32 --port /dev/ttyUSB1 --baud 460800 --before default_reset write_flash -z --flash_mode dout --flash_freq 40m --flash_size detect 0x1000 examples/default/bin/esp32-ethernet-kit-v1_0/esp_bootloader/bootloader.bin 0x8000 examples/default/bin/esp32-ethernet-kit-v1_0/partitions.bin 0x10000 examples/default/bin/esp32-ethernet-kit-v1_0/default.elf.bin
esptool.py v3.2-dev
Serial port /dev/ttyUSB1
```
Command
```
dist/tools/usb-serial/ttys.py --driver 'ftdi_sio' --vendor FTDI --model 'Dual RS232-HS' --iface-num 1
```
gives:
```
path         | driver   | vendor | model         | model_db                      | serial | ctime    | iface_num
-------------|----------|--------|---------------|-------------------------------|--------|----------|----------
/dev/ttyUSB2 | ftdi_sio | FTDI   | Dual RS232-HS | FT2232C/D/H Dual UART/FIFO IC | None   | 08:12:17 | 1        
```

maribu commented:
@gschorcht: Sorry, I forgot to add the `MOST_RECENT_PORT=1` in the cmd in the PR description. (I have added an `export MOST_RECENT_PORT=1` to my `~/.profile`, so I didn't notice when testing.)

The reasoning to have this optional is that it tat slower (launching one Python3 VM is the most time consuming part of that) and requires the UDEV python3 module to be installed. But maybe it makes sense to turn this from an opt-in to an opt-out.

maribu commented:
> BTW, such a TTY board filter could be used for all ESP32x boards that use a USB2UART bridge.

Thanks for the info. I added the filter for the boards you provided the details of as well. I also added a short comment on top of the filter in the `Makefile.include`.

maribu commented:
Regarding the `esp32-wrover-kit`: It also uses the `Dual RS232-HS`, which actually contains two UART2USB bridges. This one will also need an `--iface-num <NUM>` to pick the correct USB endpoint. Would you mind to check if `0` or `1` is the correct interface number?

gschorcht commented:
> Would you mind to check if `0` or `1` is the correct interface number?

Since `PORT_LINUX` is set to `/dev/ttyUSB0` as for the `esp32-ethernet-kit-v1_0` the `--iface_num` has to be 1.

maribu commented:
Thx :-)

I added the filters to all boards and grouped the changes to one commit for all boards using the same TTY filter.
